### PR TITLE
[UI] Refactor BuilderFlow into smaller hooks (AAP-74113)

### PR DESCRIPTION
**Pull request by skochay330**
## Description

Refactors `BuilderFlow.tsx` from a 648-line monolith into focused hooks and a presentational canvas component. No user-visible behavior change — this is a maintainability refactor for AAP-74113.

## Type of Change

- [x] Code refactoring (no functional changes)
- [x] Test coverage improvement

## Changes Made

- Slim `BuilderFlow.tsx` (~200 lines) to orchestration only
- Extract ticket hooks: `useWorkflowGraphInit`, `useExecutionStateEnrichment`, `useCanvasInteractions`
- Add supporting hooks: `useBuilderFlowExecutionContext`, `useBuilderFlowSideEffects` (grouped params)
- Extract `BuilderFlowCanvas` presentational component + CSS module
- Move stored-position detection into `useWorkflowInitialization` (removes `eslint-disable`)
- Fix pending-edge cleanup to clear stale state when panel closes or source node is removed
- Add unit + vitest-axe tests for all new hooks and `BuilderFlowCanvas`

## Out of Scope

- Visual regression baseline updates (no intentional UI change)
- Further decomposition of `useBuilderFlowSideEffects` beyond grouped params

## Related Issues

Related to AAP-74113

## How to Test

1. Open the workflow builder and verify canvas loads with existing workflows
2. Switch workflows, undo/redo, and drag nodes — positions should persist correctly
3. Enter execution view — node/edge execution overlays should appear and clear on return to edit mode
4. Drag a pending edge from a node, close the side panel — pending edge should be dropped
5. Run: `npx --prefix packages/syntara-ui vitest run packages/syntara-ui/src/routes/builder/BuilderFlow`

## Frontend Checklist

- [x] I have run `npm test` and all tests pass (in `frontend/`) — 88 tests in affected BuilderFlow/hook files
- [x] I have run `npm run lint` and code passes all checks (0 errors on changed files)
- [x] I have run `npm run tsc` and there are no type errors
- [x] I have added tests (unit / E2E) for new functionality
- [x] Accessibility has been considered (semantic HTML, labels, keyboard navigation)
- [x] No visual change — refactor only; screenshots not required

## Visual Regression

- [x] No intentional UI change — baselines not updated

## Additional Notes

`BuilderFlow` previously required `eslint-disable` for `max-lines-per-function` and `complexity`; those suppressions are removed.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 export function LoopBackEdge(props: BaseEdgeProps) {
   const { sourceX, sourceY, targetX, targetY, label, style, id, source, target, data, markerEnd, selected } = props
-  const reactFlowInstance = useReactFlow()
-  const { getNodes } = reactFlowInstance
+  const geometrySelector = useCallback(
+    (state: { nodes: Parameters<typeof selectLoopBackEdgeGeometry>[0] }) =>
+      selectLoopBackEdgeGeometry(state.nodes, source, target, sourceX, sourceY, targetX, targetY),
+    [source, target, sourceX, sourceY, targetX, targetY]
+  )
+  const geometry = useStore(geometrySelector, loopBackEdgeGeometryEqual)
```
Comment (nicklemmon) [inline frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx]: AI-assisted: This line changes real behavior. The old code called `getNodes()` one time. The new code uses `useStore` with a selector. Now the edge path updates every time a node's position or size changes. This is likely a good fix. But the PR description says "No user-visible behavior change." Please confirm this change is intended. Please also check if visual regression tests need new baselines for this edge.

**Review by nicklemmon — COMMENTED** [review]
AI-assisted review (Simplified Technical English)

This review follows the project's /frontend-review-pr checklist.

**Scope note (please confirm):** This PR's diff has code from two features. One feature is the BuilderFlow hook extraction. This matches the PR title and description. The other feature is the loop-grouping drag feature (new files: useLoopGroupPositionSync.ts, loopUtils.ts, loopBackEdgeGeometry.ts, and changes to LoopBackEdge.tsx and usePositionEventHandlers.ts). This second feature has its own open PR: #494, "move loop body nodes together when dragging loop." The first commit on this branch is titled "move entire loop together." This matches PR #494's subject. If this PR merges first, PR #494's feature merges with it, without a separate review. Please confirm this is intended. If it is not intended, rebase this branch onto `devel` after #494 merges. Or move the loop-grouping commits to #494 only.

**Findings:** two inline comments below (both are questions, not blocking bugs).

**Good parts:**
- The hook extraction is a clean, faithful move. Logic is not changed for `BuilderFlow.tsx`, `useBuilderFlowExecutionContext.ts`, `useBuilderFlowSideEffects.ts`, `useCanvasInteractions.ts`, `useWorkflowGraphInit.ts`, and `useExecutionStateEnrichment.ts`. Every prop and dependency is wired through correctly.
- Every new hook has a matching test file. `BuilderFlowCanvas` has `vitest-axe` coverage for both the idle and running-execution states.
- No new `eslint-disable`, `@ts-ignore`, `@ts-expect-error`, or `// TODO` comments. The old suppressions on `BuilderFlow.tsx` were removed, as the PR description says.
- Test coverage for the new loop-grouping logic is thorough. It covers many edge cases: nested loops, done branches, and multi-select drag.
- No unsafe `any` casts, no new `forwardRef`, no new `useContext`. Derived data uses `useMemo` where needed (e.g. `preResolvedNodes`, `resolvedPendingEdge`).

Checked against `frontend/AGENTS.md` "Common PR Mistakes" checklist: no violations found beyond the two inline notes below.

File: frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        visited.add(nodeId)
+
+        addBodyNode(nodeId)
+
+        // SECURITY: Domain-invariant check — loop body can't exceed total node count
+        if (bodyNodeIds.length > realNodes.length) break
+
+        // Follow source edges, but not edges back to the loop node (prevents circular traversal)
+        const outgoing = realEdges.filter(
+          (e) => e.source === nodeId && e.sourceHandle === EdgeHandleEnum.SOURCE && e.target !== node.id
```
Comment (nicklemmon) [inline frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts]: AI-assisted: This code follows edges with the SOURCE handle only. It does not follow the LOOP handle into a nested loop's body. Because of this, `getLoopBodyMap()` may not include the body nodes of an inner loop when it builds the outer loop's body list.

This function is now also used for the group-drag feature (in `usePositionEventHandlers.ts`) and for position sync on load (in `useLoopGroupPositionSync.ts`). A drag on an outer loop may not move an inner loop's body nodes with it.

Please test this case: a loop inside a loop, with body nodes in the inner loop. Confirm a drag on the outer loop moves all of these nodes together. If it does not, this may need a fix.

File: frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx (hunk 1 above)
Comment (nicklemmon) [inline frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx]: AI-assisted: This change makes the edge path update live, on every node measurement change. Before this change, the code called `getNodes()` one time per render and did not react to store updates on its own.

This is a real behavior change, not just a refactor. The PR description says "No user-visible behavior change." Please update the PR description to call out this fix. Please also confirm the change is intended and wanted (it looks like a good fix for edge paths not updating after a loop body node resizes).

Comment (skochay330) [reply frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx]: Stale review from a bad rebase- sorry for the confusion

File: frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts (hunk 2 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts]: Stale review from a bad rebase- sorry for the confusion

File: frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx (hunk 1 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/routes/builder/edges/LoopBackEdge.tsx]: Stale review from a bad rebase- sorry for the confusion

**Review by nicklemmon — APPROVED** [review]

File: frontend/packages/syntara-ui/src/routes/builder/BuilderFlowCanvas.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+type BuilderFlowCanvasProps = {
+  containerRef: React.RefObject<HTMLDivElement | null>
+  readOnlyProp?: boolean
+  effectiveExecutionStatus: string | null
+  isReadOnly: boolean
+  nodes: NodeType[]
+  edges: EdgeType[]
+  onNodesChange: (changes: NodeChange<NodeType>[]) => void
+  onNodeDragStart?: OnNodeDrag<NodeType>
+  onNodeDrag?: OnNodeDrag<NodeType>
```
Comment (mfrances17) [inline frontend/packages/syntara-ui/src/routes/builder/BuilderFlowCanvas.tsx:37]: this could be dead code, since `onNodeDragStart` and `onNodeDrag` are never passed in `BuilderFlow`

**Review by mfrances17 — APPROVED** [review]
👍 LGTM... one small comment but not blocking
